sql/src/main/java/io/crate/execution/engine/join/JoinOperations.java

```diff
@@ -144,15 +143,16 @@ public static Projection createJoinProjection(List<Symbol> outputs, List<Symbol>
     /**
      * Converts any implicit join conditions of the WHERE clause to explicit {@link JoinPair}.
      * Every join condition that gets to be converted is removed from the {@code splitQueries}
+     *
      * @param explicitJoinPairs The explicitJoinPairs as originally written in the query
      * @param splitQueries      The remaining queries of the WHERE clause split by involved relations
      * @return the new list of {@link JoinPair}
      */
     public static List<JoinPair> convertImplicitJoinConditionsToJoinPairs(List<JoinPair> explicitJoinPairs,
                                                                           Map<Set<QualifiedName>, Symbol> splitQueries) {
```

Comment:
as this is a public method I think it would be nice to change the signature and replace the `Map` with a `LinkedHashMap` but that's probably a different task.

Reply:
The method logic does not rely on sorted entries inside the map and thus does not require a `LinkedHashMap`.

Reply:
Oh, I thought the order is also important. ok then :)